one-pager-maker/src/pages/Edit.tsx

```diff
@@ -1,54 +1,100 @@
-import { useState } from 'react'
-import Markdown from 'react-markdown'
 import './Edit.css'
+import { useEffect, useState } from 'react'
+import Markdown from 'react-markdown'
+import { useUpdateDocumentMutation, useFetchDocumentQuery } from "../redux/document/documentsApi.ts";
+import { useNavigate, useParams } from "react-router-dom";
+import { Document, Status } from "../redux/document/documentType.ts";
+import { auth } from '../firebase';
 
 function Edit() {
-  const [markdown, setMarkdown] = useState('');
-  const toMarkdownText = (e:React.ChangeEvent<HTMLTextAreaElement>) => {
-    setMarkdown(e.target.value);
-  };
+  const uid = auth.currentUser?.uid ?? "";
+
+  const [title, setTitle] = useState('');
```

Comment:
_This is also an advice_

This code defines states for each field of `Document` data, but it would be better to define only `document` with `useState<Document | undefined>(undefined)`.
To update an only field of `document`, you can use this statement below:
```TypeScript
setDocument(prev => ({...prev, title: "new title"}));
```